### PR TITLE
Lucky Numbers in Python 

### PR DESCRIPTION
**Pull request by eksdee-sama**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Yes)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(#4772)

### CODE REVIEW

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Add markdown updates

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Few changes: 

- Add Multi-Line Comments 
- Add a beginner-friendly description at the top
- Use proper code Format 
- Remove extra lines of code and spaces
- Add sample I/O at the end of the Code
- Add Time and Space Complexity at the end of the Code   
- Add new line at the EOF

File: Python/math/Lucky_Numbers.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+1 3 7 9 13 15 21 25
+Keep eliminating the nth leftover numbers, where n is the following number in the rundown after the last surviving number. Next in this model is 9.
+One way that the application of the procedure differs from that of the Sieve of Eratosthenes is that for n being the number being multiplied on a specific pass, the first 
+number eliminated on the pass is the n-th remaining number that has not yet been eliminated, as opposed to the number 2n. That is to say, the list of numbers this sieve counts 
+through is different on each pass (for example 1, 3, 7, 9, 13, 15, 19... on the third pass), whereas in the Sieve of Eratosthenes, the sieve always counts through the entire
+original list (1, 2, 3...).
+At the point when this method has been done totally, the leftover whole numbers are the lucky numbers:
+1, 3, 7, 9, 13, 15, 21, 25, 31, 33, 37, 43, 49, 51, 63, 67, 69, 73, 75, 79, 87, 93, 99, 105, 111, 115, 127, 129, 133, 135, 141, 151, 159, 163, 169, 171, 189, 193, 195, 201, 205,
+211, 219, 223, 231, 235, 237, 241, 259, 261, 267, 273, 283, 285, 289, 297, ...)
```
Comment (HarshCasper) [inline Python/math/Lucky_Numbers.py]: Please make it short and crisp

File: Python/math/Lucky_Numbers.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        print(LuckyNos)
+        
+'''
+EX1:
+input=100
+output=[1, 3, 7, 9, 13, 15, 21, 25, 31, 33, 37, 43, 49, 51, 63, 67, 69, 73, 75, 79, 87, 93, 99]
+EX2:
+input=300
+output=[1, 3, 7, 9, 13, 15, 21, 25, 31, 33, 37, 43, 49, 51, 63, 67, 69, 73, 75, 79, 87, 93, 99, 105, 111, 115, 127, 129, 133, 135, 141, 151, 159, 163, 169, 171, 189, 193, 195,
+201, 205, 211, 219, 223, 231, 235, 237, 241, 259, 261, 267, 273, 283, 285, 289, 297]
```
Comment (HarshCasper) [inline Python/math/Lucky_Numbers.py]: Use smaller I/O samples

File: Python/math/Lucky_Numbers.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        j = 1                             #iterator variable
+        while j <= len(odds) - 1 and odds[j] <= len(odds):
+                odds = [Li for i, Li in enumerate(odds) if (i + 1) % odds[j]] #counting and adding the lucky numbers in the list
+                j += 1                    #incrementing the variable
+        return odds
+#Driver code
+if __name__ == "__main__":
+        n=int(input())                  #User input
+        LuckyNos=lucky(n)               #Function Call
+        print(LuckyNos)
```
Comment (HarshCasper) [inline Python/math/Lucky_Numbers.py]: use comments above the line of code not beside it

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: Python/math/Lucky_Numbers.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+Consistently number (every much number) in the list is dispensed with, leaving just the odd numbers:
+1 3 5 7 9 11 13 15 17 19 21 23 25
+The main number excess in the list after 1 is 3, so every third number which stays in the list (only one out of every odd various of 3) is dispensed with. The first of 
+these is 5:
+1 3 7 9 13 15 19 21 25
+The following enduring number is presently 7, so every seventh leftover number is disposed of. The first of these is 19:
+1 3 7 9 13 15 21 25
+Keep eliminating the nth leftover numbers, where n is the following number in the list after the last surviving number. Next in this model is 9. Keep repeating until the sieve
+of elimination becomes larger than the index of the number[ which is to be checked if lucky ] , then the number is called Lucky Number. If a number is eliminated by any seieve 
+then the number is not a lucky number.
```
Comment (ankitaggarwal23) [inline Python/math/Lucky_Numbers.py]: Please make it short and readable

File: Python/math/Lucky_Numbers.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        LuckyNos=lucky(n)               
+        print(LuckyNos)
+        
+'''
+EX1:
+input=100
+output=[1, 3, 7, 9, 13, 15, 21, 25, 31, 33, 37, 43, 49, 51, 63, 67, 69, 73, 75, 79, 87, 93, 99]
+EX2:
+input=150
+output=[1, 3, 7, 9, 13, 15, 21, 25, 31, 33, 37, 43, 49, 51, 63, 67, 69, 73, 75, 79, 87, 93, 99, 105, 111, 115, 127, 129, 133, 135, 141]
```
Comment (ankitaggarwal23) [inline Python/math/Lucky_Numbers.py]: Use proper display message and write it properly

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Please make it short and readable

File: Python/math/Lucky_Numbers.py (hunk 4 above)
Comment (eksdee-sama) [reply Python/math/Lucky_Numbers.py]: @ankitaggarwal23 I tried to make it more readable , but really cannot make it any shorter ,as making shorter may hamper the clarity of the problem.

File: Python/math/Lucky_Numbers.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+Every second number (all even numbers) in the list is eliminated, leaving only the odd integers:
+1 3 5 7 9 11 13 15 17 19 21 23 25
+The first number remaining in the list after 1 is 3, so every third number which remains in the list (not every multiple of 3) is eliminated. The first of these is 5:
+1 3 7 9 13 15 19 21 25
+The next surviving number is now 7, so every seventh remaining number is eliminated. The first of these is 19:
+1 3 7 9 13 15 21 25
+The next surviving number after 7 is 9, so every 9th element was meant to be deleted, but as we see 9 is greater than size of list , So we got our list of surviving numbers 
+from seive of elimination i.e. LUCKY NUMBERS upto input number.
+
+'''
```
Comment (ankitaggarwal23) [inline Python/math/Lucky_Numbers.py]:
```
'''
Write a program to print all the Lucky Numbers upto the input number.

Approach:
Lets understand with input of 25 for example.
Start with a list of numbers beginning with 1:
1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25

Every second number (all even numbers) in the list is eliminated, leaving only the odd integers:
1 3 5 7 9 11 13 15 17 19 21 23 25

The first number remaining in the list after 1 is 3, so every third number which remains in the list 
(not every multiple of 3) is eliminated. The first of these is 5:
1 3 7 9 13 15 19 21 25

The next surviving number is now 7, so every seventh remaining number is eliminated. The first of these is 19:
1 3 7 9 13 15 21 25

The next surviving number after 7 is 9, so every 9th element was meant to be deleted, but as we see 9 is greater
than size of list , So we got our list of surviving numbers from seive of elimination i.e. LUCKY NUMBERS upto 
input number.
'''
```

Rewrite it like like this,

File: Python/math/Lucky_Numbers.py [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+SAMPLE 1:-
+
+Enter a number :25
+The list of all the lucky numbers upto 25  is : [1, 3, 7, 9, 13, 15, 21, 25]
+
+SAMPLE 2:-
+
+Enter a number :150
+The list of all the lucky numbers upto 130 is : [1, 3, 7, 9, 13, 15, 21, 25, 31, 33, 37, 43, 49, 51, 63, 67, 69, 73, 75, 79, 87, 93, 99, 105, 111, 115, 127, 129]
```
Comment (ankitaggarwal23) [inline Python/math/Lucky_Numbers.py]: Remove the extra test case

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**